src/NuGet.Core/NuGet.Build.Tasks/GenerateTemporaryTargetAssembly2.cs

```diff
@@ -0,0 +1,523 @@
+        {
+        }
+
+        #endregion Constructors
+
+        //------------------------------------------------------
+        //
+        //  Public Methods
+        //
+        //------------------------------------------------------
+
+        #region Public Methods
+
+        /// <summary>
+        /// ITask Execute method
+        /// </summary>
+        /// <returns></returns>
+        public override bool Execute()
+        {
+            bool retValue = true;
+
+            // Verification
+            try
+            {
+                XmlDocument xmlProjectDoc = null;
+
+                xmlProjectDoc = new XmlDocument();
+                xmlProjectDoc.Load(CurrentProject);
+
+                //
+                // remove all the WinFX specific item lists
+                // ApplicationDefinition, Page, MarkupResource and Resource
+                //
+
+                RemoveItemsByName(xmlProjectDoc, APPDEFNAME);
+                RemoveItemsByName(xmlProjectDoc, PAGENAME);
+                RemoveItemsByName(xmlProjectDoc, MARKUPRESOURCENAME);
+                RemoveItemsByName(xmlProjectDoc, RESOURCENAME);
+
+                // Replace the Reference Item list with ReferencePath
+
+                RemoveItemsByName(xmlProjectDoc, REFERENCETYPENAME);
+                AddNewItems(xmlProjectDoc, ReferencePathTypeName, ReferencePath);
+
+                // Add GeneratedCodeFiles to Compile item list.
+                AddNewItems(xmlProjectDoc, CompileTypeName, GeneratedCodeFiles);
+
+
+                // Create a random file name
+                // This can fix the problem of project cache in VS.NET environment.
+                //
+
+                // GetRandomFileName( ) could return any possible file name and extension, but
+                // some file externsion has special meaning in MSBUILD system, such as a ".sln"
+                // means the file is a solution file with special file format. Since the temporary
+                // file is just for a project, we can use a fixed extension here, but the basic 
+                // file name is still random which can fix above VS.NET bug.
+                //
+                string randProjPath = Path.ChangeExtension(Path.GetRandomFileName(), Path.GetExtension(CurrentProject));
+
+                // Save the xmlDocument content into the temporary project file.
+                xmlProjectDoc.Save(randProjPath);
+
+                //
+                // Invoke MSBUILD engine to build this temporary project file.
+                //
+
+                Hashtable globalProperties = new Hashtable(3);
+
+                // Add AssemblyName and IntermediateOutputPath to the global property list
+                globalProperties[intermediateOutputPathPropertyName] = IntermediateOutputPath;
+                globalProperties[assemblyNamePropertyName] = AssemblyName;
+                globalProperties[originalProjectNamePropertyName] = Path.GetFileNameWithoutExtension(CurrentProject);
```

Comment:
I'm surprised to see these are all the global properties that are set. There could be any number of other global properties that were passed to the outer build that could result in this project experiencing a build break when they are dropped. If that's prior behavior of the original task, then I guess we can Won't Fix this as one of the Evils of the WPF inner-build design.

Reply:
globalProperties were used for intermediateOutputPathPropertyName, assemblyNamePropertyName. I added originalProjectNamePropertyName
Resolved won't fix due to evil design.